src/EventSubscriber/MigrateStorageSubscriber.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusGoogleAdsPlugin\EventSubscriber;
+
+use Setono\SyliusGoogleAdsPlugin\TrackingInformation\TrackingInformation;
+use Setono\SyliusGoogleAdsPlugin\TrackingInformation\TrackingInformationStorageInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * This subscriber is responsible for migrating the storage of the tracking information from the cookie to the client metadata.
+ * This subscriber should only be used when the storage is set to client_metadata
+ */
+final class MigrateStorageSubscriber implements EventSubscriberInterface
+{
+    private ?TrackingInformation $trackingInformation = null;
+
+    public function __construct(
+        private readonly TrackingInformationStorageInterface $trackingInformationStorage,
+        private readonly string $cookieName,
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::REQUEST => ['migrate', 5], // we want to run this before the StoreTrackingInformationSubscriber
+            KernelEvents::RESPONSE => 'remove',
+        ];
+    }
+
+    public function migrate(RequestEvent $event): void
+    {
+        if (!$event->isMainRequest()) {
+            return;
+        }
+
+        try {
+            $this->trackingInformation = TrackingInformation::fromCookie($event->getRequest(), $this->cookieName);
+        } catch (\Throwable) {
+            return;
+        }
+
+        $this->trackingInformationStorage->store($this->trackingInformation);
+    }
+
+    public function remove(ResponseEvent $event): void
+    {
+        if (null === $this->trackingInformation || !$event->isMainRequest()) {
+            return;
+        }
+
+        try {
+            $cookie = $this->trackingInformation->toCookie($this->cookieName, 1);
```

Comment:
Jeg antager at vi vil fjerne cookien, men det er måske ikke direkte muligt? Og at vi i stedet sætter den til at expire om 1 sekund?

Reply:
Ja, det er sådan, Symfony gør det: https://github.com/symfony/symfony/blob/7.0/src/Symfony/Component/HttpFoundation/ResponseHeaderBag.php#L222. Grunden til, at jeg ikke bruger deres metode til det er at der (vist nok) er nogle gange en cookie ikke bliver slettet, hvis man ikke angiver de samme parametre for cookien, som den blev sat med.